### ISSUE TITLE
Promotion step is failing to create a PR

### ISSUE THREAD

**Issue by PierreBtz** [jenkins-x/jx#5538]
### Summary 

Promotion step is failing to create a PR with a `422 Validation Failed [{Resource:PullRequest Field:head Code:invalid Message:}]` error message

```
error: creating pull request with arguments Title: chore: go-demo-6 to 1.0.405; Body: chore: Promote go-demo-6 to version 1.0.405; Head: *********:promote-go-demo-6-1.0.405; Base: master; Labels: ; Git Repo: https://api.github.com/repos/pierrebtz-jx/environment-jx-rocks-staging: POST https://api.github.com/repos/pierrebtz-jx/environment-jx-rocks-staging/pulls: 422 Validation Failed [{Resource:PullRequest Field:head Code:invalid Message:}]
```

### Steps to reproduce the behavior

* Follow the steps in http://vfarcic.github.io/jx/workshop-tekton.html#/15 (make sure to `jx import` the project first...

* Notes that may (or may not I didn't debug...) be important:

1. my github username contains capital letters (PierreBtz). 
2. I'm using the pierrebtz-jx organization to workaround a capital letter issue.

I found https://github.com/jenkins-x/jx/issues/3994 that looks to describe the issue (impossible to create a PR if the target repository is not a fork) but a fix was merged already for it (https://github.com/jenkins-x/jx/pull/3995)).

### Expected behavior

PR is created and the Pipeline succeeds.

### Actual behavior

PR is not created, the Pipeline fails and no promotion can happen.

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.757
jenkins x platform 2.0.1247
Kubernetes cluster v1.13.7-gke.8
kubectl            v1.14.2
helm client        Client: v2.14.0+g05811b8
git                2.21.0
Operating System   Mac OS X 10.14.6 build 18G87
```

### Jenkins type

- [x] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster

GKE created using:

```
jx create cluster gke -n jx-rocks -p $PROJECT -r us-east1 \
    -m n1-standard-4 --min-num-nodes 1 --max-num-nodes 2 \
    --default-admin-password=$PASSWORD \
    --default-environment-prefix jx-rocks --git-provider-kind github \
    --namespace $NAMESPACE --prow --tekton
```

### Operating system / Environment

```
Mac OS X 10.14.6 build 18G87
```

CC @vfarcic

**Comment by nehagup** [comment]
I faced the same issue.

**Comment by srehmanproov** [comment]
Same issue for me as well.

**Comment by hferentschik** [comment]
I have seen this happening when the "fork" of the environment repository already existed, but the information that it is an actual fork was lost. I am not quite sure how one can get into this situation atm, but apparently it is possible. A workaround for me in this situation was deleting the fork and rerunning the pipeline/promotion. When the "fork" repository does not exist a new one is created which seems to properly track from which repo it is forked.

Basically what happens is that the pull request API call against the dev repo fails since the repo which contains the change is not recognised as a fork.

**Comment by karthickvelu90** [comment]
Same issue for me as well. Any mitigation idea??

**Comment by karthickvelu90** [comment]
@hferentschik  Your way is apparently correct. Deleting the Fork repo under the owner of repository will resolve the issue.
